components/provisioning/org.wso2.carbon.identity.provisioning/src/main/java/org/wso2/carbon/identity/provisioning/listener/DefaultInboundUserProvisioningListener.java

```diff
@@ -230,9 +224,154 @@ public boolean doPreSetUserClaimValues(String userName, Map<String, String> inbo
+                log.debug("Adding domain name : " + domainName + " to user : " + userName);
+            }
+            String domainAwareName = UserCoreUtil.addDomainToName(userName, domainName);
+
+            ProvisioningEntity provisioningEntity = new ProvisioningEntity(
+                    ProvisioningEntityType.USER, domainAwareName, ProvisioningOperation.PATCH,
+                    outboundAttributes);
+
+            Map<String, String> inboundAttributes = new HashMap<>();
+            inboundAttributes.put(attributeToDelete,"");
+
+            // set the in-bound attribute list.
+            provisioningEntity.setInboundAttributes(inboundAttributes);
+
+            String tenantDomainName = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+
+            ThreadLocalProvisioningServiceProvider threadLocalServiceProvider;
+            threadLocalServiceProvider = IdentityApplicationManagementUtil
+                    .getThreadLocalProvisioningServiceProvider();
+
+            if (threadLocalServiceProvider != null) {
+
+                String serviceProvider = threadLocalServiceProvider.getServiceProviderName();
+                tenantDomainName = threadLocalServiceProvider.getTenantDomain();
+                if (threadLocalServiceProvider.getServiceProviderType() == ProvisioningServiceProviderType.OAUTH) {
+                    try {
+                        serviceProvider = ApplicationManagementService.getInstance()
+                                .getServiceProviderNameByClientId(
+                                        threadLocalServiceProvider.getServiceProviderName(),
+                                        IdentityProvisioningConstants.SP_OAUTH2_TYPE, tenantDomainName);
+                    } catch (IdentityApplicationManagementException e) {
+                        log.error("Error while provisioning", e);
+                        return true;
+                    }
+                }
+
+                // call framework method to provision the user.
+                OutboundProvisioningManager.getInstance().provision(provisioningEntity,
+                        serviceProvider, threadLocalServiceProvider.getClaimDialect(),
+                        tenantDomainName, threadLocalServiceProvider.isJustInTimeProvisioning());
+            } else {
+                // call framework method to provision the user.
+                OutboundProvisioningManager.getInstance()
+                        .provision(provisioningEntity, ApplicationConstants.LOCAL_SP,
+                                IdentityProvisioningConstants.WSO2_CARBON_DIALECT, tenantDomainName, false);
+            }
+
+            return true;
+        } finally {
+            IdentityApplicationManagementUtil.resetThreadLocalProvisioningServiceProvider();
+        }
+    }
+
+    @Override
+    public boolean doPostDeleteUserClaimValue(String userName, UserStoreManager userStoreManager)
+            throws UserStoreException {
+        return true;
```

Comment:
Let's have a default implementation for all the methods in AbstractUserOperationEventListener similar to what we have done in #976